### ISSUE TITLE
Reviews are displayed on the user's profile page only after refreshing page 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#5470]
STR:
1. Load AMO dev and submit a review for an add-on
2. Go to View my profile and check if the review is displayed

Expected result:
The review is displayed after clicking View My profile.

Actual result:
No review is available.

Notes:
The Review is available only after refreshing the page or navigation to dev hub and then back to View profile is made.
This issue is reproducible on AMO dev with FF61(Win10). 

![review showp up](https://user-images.githubusercontent.com/33448286/42157734-37dc77c4-7df7-11e8-909e-20cbd9eb6dd9.gif)

**Comment by willdurand** [comment]
Indeed, we do not reload the reviews until page gets refreshed. I am not sure it is super useful to reload them when a new review has been submitted to be honest.

**Comment by kumar303** [comment]
This is an edge case but it's worth fixing because Redux makes it easy ❤️ 

This is the fix:

```diff
diff --git a/src/amo/reducers/reviews.js b/src/amo/reducers/reviews.js
index 37433ede6..d8ff4cc87 100644
--- a/src/amo/reducers/reviews.js
+++ b/src/amo/reducers/reviews.js
@@ -264,6 +264,11 @@ export default function reviewsReducer(
       return {
         ...state,
         byId: storeReviewObjects({ state, reviews: [payload] }),
+        byUserId: {
+          ...state.byUserId,
+          // This will trigger a refresh from the server.
+          [payload.userId]: undefined,
+        },
         [payload.userId]: {
           ...state[payload.userId],
           // TODO: this should be a list of review IDs, not objects. It will
```

**Comment by ioanarusiczki** [comment]
Verified fixed on AMO dev with FF61(Win10&Android 8.0)

![no refresh page needed](https://user-images.githubusercontent.com/33448286/42501962-8fa9f138-843d-11e8-949c-c3397b63f088.gif)
